### ISSUE TITLE
Declined charge not being stored in DB, previous charge still active

### ISSUE THREAD

**Issue by Kyon147** [ohmybrew/laravel-shopify#307]
## Expected Behavior

I'm triggering the next billing plan, when a user sync products to the DB and it is over a certain amount. When the user click cancel, it assumed it would store the charge in the DB I assume and marked it as declined.

This way I can add next steps to make sure they have an active plan at the right level or constantly ask them to accept the billing to do anything or lock down parts of the app until they do.

## Current Behavior

The app is being redirected to the billing page and back to app homepage if they decline but nothing in the DB is happening. Old charge is still there and still "active", new charge which was declined is not even saved in the DB so no record of it.

![image](https://user-images.githubusercontent.com/32519106/62892887-31f25600-bd41-11e9-9e25-d65d07f349a4.png)

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Create two plans
2. Add user to first plan
3. Trigger next plan and cancel it
4. User is pushed to apps page with no decline message and nothing happens in the app to store that it has happened.

### Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* Package Version: v8.0
* Laravel Version: v5.7
* PHP Version: v7.2

**Comment by Kyon147** [comment]
I'm looking at a workaround for this but wondered, what is the best way of adding a custom method to the Shop Model? 

I've added a "upgrade_to_plan_id" column in the Shop table. Then will use this to see if the plan_id and this match in a custom middle ware. If not redirect them to the billing screen. 

This seems like the best way around the declining as I think it changed recently?

**Comment by ohmybrew** [comment]
@Kyon147 If someone clicks decline on Shopify for the charge, to my knowledge there is no flag or anything sent back for us to detect.

**Comment by Kyon147** [comment]
@ohmybrew no worries, was not 100% sure! 

In the end I've added a migration which adds a field to the Shop model called `upgrade_to_plan_id` this then gets updated when the products are sync to the database. Depending on the amount it sets this to the plan it needs to be. This field could be used for any logic though.

Then I have made a middleware which checks to make sure `plan_id `and `upgrade_to_plan_id `are the name or `upgrade_to_plan_id `is null.

If they are not the same or null, then it redirects using your billing route to trigger the plan. I've also added a method for the blade view so that it removes certain buttons and replaces them with "upgrade".

This seemed like the best way to do it and thought I'd outline in in case someone else wants to do something similar.
